ecl/eclagent/eclagent.cpp

```diff
@@ -3177,14 +3190,30 @@ void EclAgent::abortMonitor()
             if (queryWorkUnit()) // could be exiting
                 isAborting = queryWorkUnit()->aborting();
         }
-        if (isAborting) {
+        if (isAborting)
+        {
             DBGLOG("Abort detected");
             while (abortmonitor->sem.wait(ABORT_DEADMAN_INTERVAL*1000))
                 if (abortmonitor->stopping)
                     return; // stopped in time
             IERRLOG("EclAgent failed to abort within %ds - killing process",ABORT_DEADMAN_INTERVAL);
             break;
         }
+        if (checkCostInterval <= waittime)
+        {
+            __int64 totalCost = aggregateCost(queryWorkUnit());
+            if (totalCost > abortmonitor->guillotineCost)
```

Comment:
can guillotineCost be unset / 0 / disabled ?
If so, the checkCostInterval and checking should be disabled.